src/components/CertificateViewer.tsx

```diff
@@ -39,8 +38,13 @@ export const CertificateViewer: FunctionComponent<CertificateViewerProps> = ({
   emailSendingState,
   handleSendCertificate,
 }) => {
-  const tokenId = getDocumentId(document);
-  const tokenRegistryAddress = getTokenRegistryAddress(document);
+  let tokenId = "";
+  try {
+    tokenId = `0x${utils.getAssetId(document)}`;
+  } catch (e) {
+    trace(e);
```

Comment:
not sure what happens to the user experience when it gets to this path, but can you check it out and if its unpleasant make a story to fix that?